### ISSUE TITLE
New gatsby-cli installation on ubuntu (wsl2) shows errors when running the command `gatsby` in terminal

### ISSUE THREAD

**Issue by ManiMuridi** [gatsbyjs/gatsby#34088]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

I just recently installed gatsby-cli on ubuntu 20 and every time I run the command `gatsby` on the command line it shows error on multiple areas. I run the same command `gatsby --verbose` as advised by gatsby message but it produces the exact same error. Here is a screenshot:

![image](https://user-images.githubusercontent.com/1406117/143757363-544c4086-53b7-4334-8ed1-a43f25a1ee5b.png)

this also prevents me from setting up projects using gatsby templates. the command throws an error when running gatsby install with a template url.

### Reproduction Link

https://gatsby.dev/reproduction

### Steps to Reproduce

1. npm i -g gastby-cli
2. gatsby
3. gatsby --verbose

### Expected Result

create a new gatsby project without errors.

### Actual Result

![image](https://user-images.githubusercontent.com/1406117/143759130-c28cf693-52d6-42d0-b616-d04525778028.png)

### Environment

```shell
System:
    OS: Linux 5.10 Ubuntu 20.04.3 LTS (Focal Fossa)
    CPU: (16) x64 AMD Ryzen 9 5900HX with Radeon Graphics
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 16.13.0 - ~/.nvm/versions/node/v16.13.0/bin/node
    Yarn: 1.22.15 - ~/.nvm/versions/node/v16.13.0/bin/yarn
    npm: 8.1.4 - ~/.nvm/versions/node/v16.13.0/bin/npm
  npmGlobalPackages:
    gatsby-cli: 4.2.0
```

**Comment by wardpeet** [comment]
Hi, it's a problem with our reporter. The cli uses console.error and thus these bits are formatted that way. It's something we need to solve but for now you can ignore it.

**Comment by jongwooo** [comment]
Hi @ManiMuridi 

I hope everything is fine with you.

Run the following command from the command line. This will start up the interactive prompt to help you create a new Gatsby site.

```shell
gatsby new
```

Please refer to the following link.
[https://www.gatsbyjs.com/docs/tutorial/part-1/](https://www.gatsbyjs.com/docs/tutorial/part-1/)

**Comment by Omar8345** [comment]
_**I had put this in my plans, will try to make PR to fix it!**_

**Comment by Omar8345** [comment]
**@ManiMuridi, dear this is not an error, Gatsby expects that you input a command like the following:**

`gatsby <command> [options]`

So, it is not expected to get that... So if you need help, `gatsby help`... Hope this helps!

**Comment by AndrewSerra** [comment]
I can try to make the error more descriptive if this is still needed. @Omar8345 are you adding anything currently? Instead of having just an "unhandled error", I can add that there should be arguments given and run "gatsby help" to see the options available. If nobody is currently doing it I'll go ahead and create a PR

**Comment by Omar8345** [comment]
Yeah, that seems to be the **best** solution possible, go ahead as that's more **better**! :)

_I won't be stealing your solution, you do the PR 😊_

**Comment by AndrewSerra** [comment]
Sounds good @Omar8345 :)

**Comment by AndrewSerra** [comment]
Trying to set up the gatsby repo by running `yarn run bootstrap` and it's giving some python errors. I have version 3.8 and it wants 2.6 > up to < 3.0. I tried looking into discord and other issues but couldn't find anything. I have an apple m1 and running macOS Monterey. Is there anywhere I can check out for this problem so I can complete the PR for this issue?

**Comment by AndrewSerra** [comment]
Are there any specific error code meanings so I can add one? That way I can attach a message to it as well. It took me a while to go through the code and understand how things interact.

**Comment by noxtstudio** [comment]
Hi you can type `gatsby new <PROJ_NAME>` and this will make a new project in your dir of choosing something like `/downloads` as an example and this will default to gatsbys base proj as seen here -> https://github.com/gatsbyjs/gatsby/tree/master/starters/default

**Comment by Rupesh59** [comment]
> ### Preliminary Checks
> * [x]  This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
> * [x]  This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions
> 
> ### Description
> I just recently installed gatsby-cli on ubuntu 20 and every time I run the command `gatsby` on the command line it shows error on multiple areas. I run the same command `gatsby --verbose` as advised by gatsby message but it produces the exact same error. Here is a screenshot:
> 
> ![image](https://user-images.githubusercontent.com/1406117/143757363-544c4086-53b7-4334-8ed1-a43f25a1ee5b.png)
> 
> this also prevents me from setting up projects using gatsby templates. the command throws an error when running gatsby install with a template url.
> 
> ### Reproduction Link
> https://gatsby.dev/reproduction
> 
> ### Steps to Reproduce
> 1. npm i -g gastby-cli
> 2. gatsby
> 3. gatsby --verbose
> 
> ### Expected Result
> create a new gatsby project without errors.
> 
> ### Actual Result
> ![image](https://user-images.githubusercontent.com/1406117/143759130-c28cf693-52d6-42d0-b616-d04525778028.png)
> 
> ### Environment
> ```shell
> System:
>     OS: Linux 5.10 Ubuntu 20.04.3 LTS (Focal Fossa)
>     CPU: (16) x64 AMD Ryzen 9 5900HX with Radeon Graphics
>     Shell: 5.0.17 - /bin/bash
>   Binaries:
>     Node: 16.13.0 - ~/.nvm/versions/node/v16.13.0/bin/node
>     Yarn: 1.22.15 - ~/.nvm/versions/node/v16.13.0/bin/yarn
>     npm: 8.1.4 - ~/.nvm/versions/node/v16.13.0/bin/npm
>   npmGlobalPackages:
>     gatsby-cli: 4.2.0
> ```
> 
> ### Config Flags
> _No response_

Delete all logs about tool and reinstall again it would help

**Comment by PranshuSrivastava** [comment]
Hi, I just read the issue and the comments associated with it.  From what I understand, we are just looking to add a more descriptive message in the error so that people can understand better what command needs to be entered. Is that right? If no one is working on this, I am willing to take this up. @AndrewSerra @Omar8345

**Comment by jairajsahgal** [comment]
> Are there any specific error code number meanings so I can add one? They don't seem to be incrementing so I wasn't sure. That way I can attach a message to it as well. It took me a while to go through the code and understand how things interact.

Did you added those messages?

**Comment by nikunj-kohli** [comment]
I would like to contribute here, I request you to assign me

**Comment by LipsaDas0710** [comment]
Hi! I'd like to work on this issue. Could you please assign it to me?
Thank you!

**Comment by LeonxLJX** [comment]
Hi! I'd like to work on this issue. Please let me know if I can be assigned. Thanks!

**Comment by LeonxLJX** [comment]
Hi! I'd like to fix gatsby-cli install errors on ubuntu (wsl2). Plan: reproduce, fix the install path, verify. May I be assigned?

**Comment by emmron** [comment]
@LeonxLJX I opened #39675 with a small master-based fix and CLI integration regression. It sends no-command help (including `gatsby --verbose`) through the normal log path while keeping invalid commands and options as errors. Happy to incorporate any findings from your reproduction.
